lib/ansible/modules/cloud/azure/azure_rm_cosmosdbaccount_facts.py

```diff
@@ -0,0 +1,418 @@
+from ansible.module_utils.common.dict_transformations import _camel_to_snake
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.mgmt.cosmosdb import CosmosDB
+    from msrest.serialization import Model
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMDatabaseAccountFacts(AzureRMModuleBase):
+    def __init__(self):
+        # define user inputs into argument
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str'
+            ),
+            tags=dict(
+                type='list'
+            )
+        )
+        # store the results of the module operation
+        self.results = dict(
+            changed=False
+        )
+        self.mgmt_client = None
+        self.resource_group = None
+        self.name = None
+        self.tags = None
+        super(AzureRMDatabaseAccountFacts, self).__init__(self.module_arg_spec, supports_tags=False)
+
+    def exec_module(self, **kwargs):
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+        self.mgmt_client = self.get_mgmt_svc_client(CosmosDB,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        if self.name is not None:
+            self.results['accounts'] = self.get()
+        else:
+            self.results['accounts'] = self.list_by_resource_group()
+        return self.results
+
+    def get(self):
+        response = None
+        results = []
+        try:
+            response = self.mgmt_client.database_accounts.get(resource_group_name=self.resource_group,
+                                                              account_name=self.name)
+            self.log("Response : {0}".format(response))
+        except CloudError as e:
+            self.log('Could not get facts for Database Account.')
+
+        if response and self.has_tags(response.tags, self.tags):
+            results.append(self.format_response(response))
+
+        return results
+
+    def list_by_resource_group(self):
+        response = None
+        results = []
+        try:
+            response = self.mgmt_client.database_accounts.list_by_resource_group(resource_group_name=self.resource_group)
+            self.log("Response : {0}".format(response))
+        except CloudError as e:
+            self.log('Could not get facts for Database Account.')
+
+        if response is not None:
+            for item in response:
+                if self.has_tags(item.tags, self.tags):
+                    results.append(self.format_response(item))
+
+        return results
+
+    def format_response(self, item):
+        d = item.as_dict()
+        d = {
+            'id': d.get('id', None),
+            'resource_group': self.resource_group,
```

Comment:
What if there is no self.resource_group set.

Reply:
now made resource_group not required and changed the code